src/lib/quoteEngine.js

```diff
@@ -246,11 +162,9 @@ const quoteEngine = (store, productId, amount, period, coverAsset) => {
     };
   });
 
-  const customPoolIdPriorityFixedPrice = selectProductPriorityPoolsFixedPrice(store, productId);
-
-  const allocations = customPoolIdPriorityFixedPrice
-    ? customAllocationPriorityFixedPrice(amountToAllocate, poolsData, customPoolIdPriorityFixedPrice)
-    : calculateOptimalPoolAllocation(amountToAllocate, poolsData, product.useFixedPrice);
+  const customPoolIdPriorityFixedPrice = selectProductPriorityPoolsFixedPrice(store, productId) || [];
+  const poolsInPriorityOrder = sortPools(poolsData, customPoolIdPriorityFixedPrice);
+  const allocations = calculatePoolAllocations(amountToAllocate, poolsInPriorityOrder);
 
   const poolsWithPremium = Object.keys(allocations).map(poolId => {
     poolId = parseInt(poolId);
```

Comment:
how we displayed the ordering of the pools get jumbled because we do `Object.keys(allocations)`.

its best if `calculatePoolAllocations` returns an array instead of an object, since all we do on the allocation object is iterate over it anyway and an array would preserve the ordering.

@duxxud could you add this as a task on the ticket that you'll create?